osquery/process/windows/process.cpp

```diff
@@ -288,23 +292,24 @@ std::shared_ptr<PlatformProcess> PlatformProcess::launchExtension(
     return std::shared_ptr<PlatformProcess>();
   }
 
-  auto ext_path = fs::path(exec_path);
+  auto ext_path = fs::path(wexec_path);
 
   // We are autoloading a Python extension, so pass off to our helper
-  if (ext_path.extension().string() == ".ext") {
-    return launchTestPythonScript(
-        std::string(mutable_argv.begin(), mutable_argv.end()));
+  if (ext_path.extension().wstring() == L".ext") {
+    return launchTestPythonScript(wstringToString(
+        std::wstring(mutable_argv.begin(), mutable_argv.end())));
   } else {
-    auto status = ::CreateProcessA(exec_path.c_str(),
-                                   mutable_argv.data(),
-                                   nullptr,
-                                   nullptr,
-                                   TRUE,
-                                   IDLE_PRIORITY_CLASS,
-                                   nullptr,
-                                   nullptr,
-                                   &si,
-                                   &pi);
+    auto status =
+        ::CreateProcess(NULL,
```

Comment:
Why is the `wexec_path` no longer used here? If it is not needed then we should use `nullptr` instead of `NULL`.

Reply:
This is an excellent question. `CreateProcess(appName, commandLine, ...)` behaves differently depending on if only appName is provided, only commandLine is provided, or both appName and commandLine are provided. If only commandLine is provided (this case), the appName is argv[0] from the commandLine. If both appName and commandLine are provided, appName is argv[0], and commandLine is argv[0..n]. That is, appName must still be provided as the first token in commandLine. So, in this code, the first parameter does not need to be provided. I will update so this is `nullptr` vs `NULL`.

Reference 1: https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessw
Reference 2: https://stackoverflow.com/questions/1135784/createprocess-doesnt-pass-command-line-arguments